src/detect-engine-threshold.c

```diff
@@ -458,17 +435,9 @@ static int ThresholdHandlePacketHost(Host *h, Packet *p, const DetectThresholdDa
                 }
             } else {
                 DetectThresholdEntry *e = DetectThresholdEntryAlloc(td, p, sid, gid);
-                if (e == NULL) {
-                    break;
-                }
-
-                e->tv_sec1 = p->ts.tv_sec;
-                e->current_count = 1;
+                *new_tsh = e;
```

Comment:
The local `e` can be removed ... just assign to `*new_tsh` instead (here, and a few other places).

Reply:
Okay.